src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -228,17 +228,11 @@ public async Task Skip(long len, bool async)
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public sbyte ReadSByte()
-        {
-            Debug.Assert(sizeof(sbyte) <= ReadBytesLeft);
-            return (sbyte)Buffer[ReadPosition++];
-        }
+            => Read<sbyte>();
```

Comment:
A thought: aren't you a little worried that generic `Read<T>()` will be slower than a hand-written `ReadSByte()` (and all the others)? `Read<T>()` calls two `Unsafe` operations, whereas the manual version that was here does nothing...

We could benchmark this to find out. If the goal is to introduce bounds checking, we can simply convert the previous `Debug.Assert()` into runtime checks.

Reply:
As before, I would like to have a common method to read everything and to throw.